InMobiAdapter/src/main/java/com/chartboost/mediation/inmobiadapter/InMobiAdapter.kt

```diff
@@ -512,16 +536,30 @@ class InMobiAdapter : PartnerAdapter {
                 return Result.failure(ChartboostMediationAdException(ChartboostMediationError.CM_LOAD_FAILURE_INVALID_PARTNER_PLACEMENT))
             }
 
-            return suspendCoroutine { continuation ->
-                InMobiInterstitial(
+            return suspendCancellableCoroutine { continuation ->
+                inMobiInterstitialAds[request.identifier] = InMobiInterstitial(
                     context,
                     placement,
                     buildFullScreenAdListener(
                         request = request,
                         partnerAdListener = partnerAdListener,
                         continuation = continuation
                     )
-                ).load()
+                )
+
+                inMobiInterstitialAds[request.identifier]?.load() ?: run {
```

Comment:
you already new'd one up. why retrieve it? that way you get rid of an entire branch.

Reply:
Yeah, let me compact this and see how it goes. 🤞🏼

Reply:
Oh, right. We need to retrieve the InMobiInterstitial to load it. Otherwise, we'd be doing what was causing the issue. 👀

Reply:
are you saying the compiler was automatically optimizing it away? lol xD
maybe try the suggestion above.... maybe.....

Reply:
Will give it a go. 👍🏼